docs/src/pages/[platform]/theming/dark-mode/examples/CustomDarkMode.jsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
```

Comment:
Here's perhaps the weirdest part of this PR -- I created `*.jsx` files based off the existing `*.tsx` examples. They do not get used however and only exists for the code snippet.

This was because TypeScript will complain if you try to use `React.useState('system')` with 

```
Type 'string' is not assignable to type 'ColorMode'
```

Reply:
Any suggestions are appreciated 🙏

Reply:
Talked with @calebpollman, we'll just go ahead and hard code the JS snippets instead of creating `.jsx` files. We can decide on a more scalable approach as we make more JS/TS tabs, but for now let's keep it simple for now.